### ISSUE TITLE
Help needed: Checking compatibility with Ideapad Slim 7 14ITL05

### ISSUE THREAD

**Issue by htv04** [esno/yoga-bios-unlock#38]
First off: Thanks for making this! I had lost all hope when it came to unlocking the BIOS of my Ideapad Slim 7, but it seems that my device's Yoga Slim 7 cousin conveniently has a solution.

Looking at the writeup on the BIOS unlock, the BIOSs of the two systems look very similar, as they both use InsydeH2O. I'd like to try using the same steps to unlock my system's BIOS. The biggest difference I can find is that the BIOS versioning is different, prefixed with "DM" instead of my system's "FN." Also, my system has system has an Intel processor instead of AMD.

However, I would rather avoid writing to and reading random ports on my system without verifying the port ranges are similar between these systems. If it wouldn't be too much trouble, would you mind providing the output of `cat /proc/ioports` on your system? If the ports are similar enough, I'll give the script a go and report back on what happens.

**Comment by esno** [comment]
hi, sure here's the `/proc/ioports`
```
# cat /proc/ioports 
0000-0cf7 : PCI Bus 0000:00
  0000-001f : dma1
  0020-0021 : pic1
  0040-0043 : timer0
  0050-0053 : timer1
  0060-0060 : keyboard
  0061-0061 : PNP0800:00
  0062-0062 : PNP0C09:00
    0062-0062 : EC data
  0064-0064 : keyboard
  0066-0066 : PNP0C09:00
    0066-0066 : EC cmd
  0070-0071 : rtc0
  0080-008f : dma page reg
  00a0-00a1 : pic2
  00c0-00df : dma2
  00f0-00ff : fpu
    00f0-00fe : PNP0C04:00
  0400-04cf : pnp 00:03
    0400-0403 : ACPI PM1a_EVT_BLK
    0404-0405 : ACPI PM1a_CNT_BLK
    0408-040b : ACPI PM_TMR
    0420-0427 : ACPI GPE0_BLK
  04d0-04d1 : pnp 00:03
  04d6-04d6 : pnp 00:03
  0800-0800 : ACPI PM2_CNT_BLK
  0b00-0b08 : piix4_smbus
  0b20-0b28 : piix4_smbus
  0c00-0c01 : pnp 00:03
  0c14-0c14 : pnp 00:03
  0c50-0c52 : pnp 00:03
  0c6c-0c6c : pnp 00:03
  0c6f-0c6f : pnp 00:03
  0cd0-0cdb : pnp 00:03
0cf8-0cff : PCI conf1
0d00-ffff : PCI Bus 0000:00
  1000-1fff : PCI Bus 0000:03
    1000-10ff : 0000:03:00.0
```
without any warranties ^^

**Comment by htv04** [comment]
I checked the ports and it does indeed seem the port ranges are pretty similar. I actually looked up more info on I/O ports and it seems ports 0x7n are the standard for RTC communication.

However, I decided not to go through with the BIOS unlock. My main reason for unlocking would be to use the S3 sleep feature that's been disabled by default, but I did some research and found that the S3 implementation on these laptops can actually be pretty broken, and Linux's S2Idle provides similar results anyway. Sorry for wasting your time on this, I just don't want to risk it, especially because I've been using this laptop for college.

I did still test the read capability of the script and modified the default value to read some other bytes. It looks like some of the values are similar to those shown in the chart on the README, but a lot of them appear to be 0x00. Still, if anyone else with an Intel-based Yoga Slim 7 wants to give it a go, it may be worth a try. For now, I'll close the issue.
